Headerless xml parsing fix

lmaxyz opened the pull request:
Found out some protocols are not available because of parser.
That patch fixes parsing for headerless xml files.

Also I added couple of tests.

elinorbgr commented on wayland-scanner/tests/scanner_assets/test-headerless-protocol.xml:
```diff
@@ -0,0 +1,252 @@
+<protocol name="test-protocol">
```
Given this test file is only used to ensure the scanner doesn't choke on it, I don't think it needs to be as big as the main one, it can be much more minimalistic I'd say.

lmaxyz replied:
It's done, thanks.

elinorbgr commented:
LGTM, could you just add an entry to the changelogs of wayland-scanner and wayland-protocols-plasma to reflect those changes ?

Also, it looks like rustfmt is unhappy in the CI

lmaxyz commented:
> LGTM, could you just add an entry to the changelogs of wayland-scanner and wayland-protocols-plasma to reflect those changes ?
> 
> Also, it looks like rustfmt is unhappy in the CI

Fixed fmt issues and added changes to changelogs.

lmaxyz commented:
@elinorbgr can you check it pls? Is it ok now?

elinorbgr reviewed (approved):
LGTM yes, thanks!
